api/src/main/java/org/openmrs/layout/name/NameSupport.java

```diff
@@ -34,9 +46,99 @@ public static NameSupport getInstance() {
 		}
 	}
 	
+	public static NameSupport getInstance(String globalProperty) {
```

Comment:
@IamMujuziMoses, why would you want to _overload_ than _override_?

Reply:
Also why do you have to provide a `globalProperty` argument, instead of use it as a public constant defined somewhere? Is it that you can properly cover this feature change in tests?

You'll want to make sure the changes you introduce here are backwards compatible with what you did on this other PR [here](https://github.com/openmrs/openmrs-module-webservices.rest/pull/586/files#diff-5596912f07778493390b06a3dab40f85c44770fa9d9d067c1cbf4f79d6c5bd58R44), i.e ` getInstance()` should be argument-less.

Reply:
right, the global property should be a public constant, okay, let me fix that

Reply:
@Ruhanga, overriding seems to seems to cause a chain reaction of errors from other classes that depend on ```NameSupport``` and ```PersonName```